File: binding-csharp/runtime/src/main/csharp/Etch/Support/AuthExceptionSerializer.cs
// $Id$
// 
// Copyright 2007-2008 Cisco Systems Inc.
// 
// Licensed under the Apache License, Version 2.0 (the "License"); you may not
// use this file except in compliance with the License. You may obtain a copy
// of the License at
//  
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
// License for the specific language governing permissions and limitations
// under the License.

using System;
using Etch.Msg;

namespace Etch.Support
{
    public class AuthExceptionSerializer : ImportExportHelper
    {
        private const String FIELD_NAME = "msg";

        public static void Init(XType type, Class2TypeMap class2type)
        {
            Field field = type.GetField(FIELD_NAME);
            class2type.Add(typeof(_Etch_AuthException), type);
            type.SetComponentType(typeof(_Etch_AuthException));
            type.SetImportExportHelper(new AuthExceptionSerializer(type, field));
            type.PutValidator(field, Validator_string.Get(0));
            type.Lock();
        }

        public AuthExceptionSerializer(XType type, Field field)
        {
            this.type = type;
            this.field = field;
        }

        private readonly XType type;

        private readonly Field field;

        public override StructValue ExportValue(ValueFactory vf, Object value)
        {
            StructValue sv = new StructValue(type, vf);
            sv.Add(field, value.ToString());
            return sv;
        }

        public override Object ImportValue(StructValue sv)
        {
            sv.CheckType(type);
            return new _Etch_AuthException((String)sv.Get(field));
        }
    }
}

File: binding-python/runtime/src/main/python/etch/binding/support/AuthExceptionSerializer.py
"""
#  Licensed to the Apache Software Foundation (ASF) under one   *
#  or more contributor license agreements.  See the NOTICE file *
#  distributed with this work for additional information        *
#  regarding copyright ownership.  The ASF licenses this file   *
#  to you under the Apache License, Version 2.0 (the            *
#  "License"); you may not use this file except in compliance   *
#  with the License.  You may obtain a copy of the License at   *
#                                                               *
#    http://www.apache.org/licenses/LICENSE-2.0                 *
#                                                               *
#  Unless required by applicable law or agreed to in writing,   *
#  software distributed under the License is distributed on an  *
#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
#  KIND, either express or implied.  See the License for the    *
#  specific language governing permissions and limitations      *
#  under the License. 
"""
from __future__ import absolute_import
from ..msg.Field import *
from ..msg.ImportExportHelper import *
from ..msg.StructValue import *
from ..msg.Type import *
from .Class2TypeMap import *
from .Validator_string import *
from ._Etch_AuthException import *

class AuthExceptionSerializer(ImportExportHelper):
    
    FIELD_NAME = "msg"
    
    @staticmethod
    def init(typ, class2type):
        field = typ.getField(AuthExceptionSerializer.FIELD_NAME)
        class2type[_Etch_AuthException] = typ
        typ.setComponentType(_Etch_AuthException)
        typ.setImportExportHelper(AuthExceptionSerializer(typ, field))
        typ.putValidator(field, Validator_string.get(0))
        typ.lock()
    
    def __init__(self, typ, field):
        self.__type  = typ
        self.__field = field
    
    def exportValue(self, value):
        sv = StructValue(self.__type)
        sv[field] = repr(value)
        return sv
    
    def importValue(self, structValue):
        structValue.checkType(self.__type)
        return _Etch_AuthException(structValue[self.__field])

        
